header-rewrite-filter/header_processor.cc

```diff
@@ -624,11 +685,11 @@ namespace HeaderRewriteFilter {
         val.set_string_value(value);
 
         // we have to erase the key if it exists, otherwise map insert will do nothing
```

Comment:
Did you verify this? Just asking because it surprises me, I would have expected `insert` to overwrite existing values

Reply:
Yeah I found it odd too -- I assumed it would replace the value but when I tested it out, I was still getting back the same value. Confirmed it in the code [here](https://chromium.googlesource.com/external/github.com/google/protobuf/+/8d5d7cc6d03b00fcb7162525100fbd4cc0d1929d/src/google/protobuf/map.h#240).